### DISCUSSION TITLE
Wait for telegram for a particular group address?

### DISCUSSION THREAD

**Discussion by DengelFred** [XKNX/xknx#1552] [General]
Hi!
I use python and would like to wait for a telegram for a particular group address, ideally with some kind of timeout. How should I do this? Unfortunately I did not found any example..
Thanks a lot!
Thomas

**Comment by farmio** [comment]
Hi 👋! 

You could create a callback on the `TelegramQueue` and match the Telegram objects `telegram.source_address` there. See https://github.com/XKNX/xknx/blob/main/examples/example_telegram_monitor.py for the callback instantiation. 

Or use a `Sensor` or `BinarySensor` device objects callback `device_updated_cb`. https://github.com/XKNX/xknx/blob/main/examples/example_sensor.py

In your application code you could use asyncio primitives for the timeout handling. Like awaiting an `asyncio.Event`, that you `set()` in the callback, in an `async with asyncio.timeout(seconds):` block. 
Maybe build a helper class - something like our `ValueReader` without the GroupValueRead request sending. https://github.com/XKNX/xknx/blob/main/xknx/core/value_reader.py
https://github.com/XKNX/xknx/blob/main/examples/example_value_reader.py
